File: flutter_app/lib/models/order_model.dart
class Order {
  final String id;
  final String customerId;
  final String productId;
  final int quantity;
  final DateTime orderDate;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? productName;
  final double? unitPrice;

  Order({
    required this.id,
    required this.customerId,
    required this.productId,
    required this.quantity,
    required this.orderDate,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.productName,
    this.unitPrice,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'],
      customerId: json['customer_id'],
      productId: json['product_id'],
      quantity: json['quantity'],
      orderDate: DateTime.parse(json['order_date']),
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
      productName: json['product_name'],
      unitPrice: json['unit_price'] != null
          ? double.parse(json['unit_price'].toString())
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'customer_id': customerId,
      'product_id': productId,
      'quantity': quantity,
      'order_date': orderDate.toIso8601String(),
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'product_name': productName,
      'unit_price': unitPrice,
    };
  }
}

File: backend/models/orderModel.js
import { getClient } from '../config/db.js';

export const createOrder = async (orderData) => {
  const client = getClient();
  const { customer_id, product_id, quantity, order_date } = orderData;
  
  const query = `
    INSERT INTO orders (customer_id, product_id, quantity, order_date)
    VALUES ($1, $2, $3, $4)
    RETURNING *
  `;
  
  const result = await client.query(query, [customer_id, product_id, quantity, order_date]);
  return result.rows[0];
};

export const findOrdersByCustomer = async (customerId, month = null) => {
  const client = getClient();
  let query = `
    SELECT o.*, p.name as product_name, p.unit_price
    FROM orders o
    JOIN products p ON o.product_id = p.id
    WHERE o.customer_id = $1
  `;
  const params = [customerId];
  
  if (month) {
    query += ` AND DATE_TRUNC('month', o.order_date) = $2`;
    params.push(`${month}-01`);
  }
  
  query += ' ORDER BY o.order_date DESC';
  
  const result = await client.query(query, params);
  return result.rows;
};

export const findOrderById = async (id) => {
  const client = getClient();
  const query = `
    SELECT o.*, p.name as product_name, p.unit_price
    FROM orders o
    JOIN products p ON o.product_id = p.id
    WHERE o.id = $1
  `;
  const result = await client.query(query, [id]);
  return result.rows[0];
};

export const updateOrderStatus = async (id, status) => {
  const client = getClient();
  const query = `
    UPDATE orders 
    SET status = $1
    WHERE id = $2
    RETURNING *
  `;
  const result = await client.query(query, [status, id]);
  return result.rows[0];
};
